packages/donor/src/screens/bookDonation/BookDonationScreenContainer.tsx

```diff
@@ -34,22 +40,53 @@ export function BookDonationScreenContainer({
   }
 
   const onSlotSelected = (donationSlot: DonationSlotToBook) => {
-    appointmentToBookStore.setAppointmentToBook(donationSlot);
-    if (isLoggedIn) {
-      navigate(MainNavigationKeys.Questionnaire);
+    setSelectedSlot(donationSlot);
+    const currentDonor = appState.donor;
+    const timeDifference = currentDonor?.lastCompletedDonationTimeMillis
+      ? DateUtils.getNumberOfDaysBetweenDates(
+          donationSlot.donationStartTimeMillis,
+          currentDonor.lastCompletedDonationTimeMillis
+        )
+      : MIN_DAYS_BETWEEN_DONATIONS + 1;
+
+    if (timeDifference < MIN_DAYS_BETWEEN_DONATIONS) {
+      setShowWarningPopup(true);
     } else {
-      navigate(MainNavigationKeys.Register);
+      continueDonation();
     }
   };
 
+  const continueDonation = () => {
+    if (selectedSlot) {
```

Comment:
Could `selectedSlot` be undefined?
I think we should throw in such cases (as this should never happen).

What do you think?